cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/ApacheKafkaAclService.java

```diff
@@ -0,0 +1,490 @@
+import org.apache.kafka.common.acl.AccessControlEntryFilter;
+import org.apache.kafka.common.acl.AclBinding;
+import org.apache.kafka.common.acl.AclBindingFilter;
+import org.apache.kafka.common.acl.AclOperation;
+import org.apache.kafka.common.acl.AclPermissionType;
+import org.apache.kafka.common.resource.PatternType;
+import org.apache.kafka.common.resource.ResourcePattern;
+import org.apache.kafka.common.resource.ResourcePatternFilter;
+import org.apache.kafka.common.resource.ResourceType;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class ApacheKafkaAclService {
+
+  private static final long TIME_OUT_SECS_FOR_ACLS = 5;
+
+  private Environment env;
+
+  private ClusterApiUtils clusterApiUtils;
+
+  public ApacheKafkaAclService(ClusterApiUtils clusterApiUtils) {
+    this.clusterApiUtils = clusterApiUtils;
+  }
+
+  public ApacheKafkaAclService() {}
+
+  public ApacheKafkaAclService(Environment env) {
+    this.env = env;
+  }
+
+  public synchronized Set<Map<String, String>> loadAcls(
+      String environment, KafkaSupportedProtocol protocol, String clusterName) throws Exception {
+    log.info("loadAcls {} {}", environment, protocol);
+    Set<Map<String, String>> acls = new HashSet<>();
+
+    AdminClient client = clusterApiUtils.getAdminClient(environment, protocol, clusterName);
+    if (client == null) {
+      throw new Exception("Cannot connect to cluster.");
+    }
+
+    try {
+      AclBindingFilter aclBindingFilter = AclBindingFilter.ANY;
+      DescribeAclsResult aclsResult = client.describeAcls(aclBindingFilter);
+
+      aclsResult
+          .values()
+          .get(TIME_OUT_SECS_FOR_ACLS, TimeUnit.SECONDS)
+          .forEach(
+              aclBinding -> {
+                filterAndUpdateAclBindings(acls, aclBinding);
+              });
+    } catch (Exception e) {
+      log.error("Error " + e.getMessage());
+    }
+
+    return acls;
+  }
+
+  private static void filterAndUpdateAclBindings(
+      Set<Map<String, String>> acls, AclBinding aclBinding) {
+    if (aclBinding.pattern().patternType().name().equals(AclPatternType.LITERAL.value)) {
+      Map<String, String> aclbindingMap = new HashMap<>();
+      aclbindingMap.put("host", aclBinding.entry().host());
+      aclbindingMap.put("principle", aclBinding.entry().principal());
+      aclbindingMap.put("operation", aclBinding.entry().operation().toString());
+      aclbindingMap.put("permissionType", aclBinding.entry().permissionType().toString());
+      aclbindingMap.put("resourceType", aclBinding.pattern().resourceType().toString());
+      aclbindingMap.put("resourceName", aclBinding.pattern().name());
+
+      if (!aclBinding.pattern().resourceType().toString().equals("CLUSTER")) {
+        if (aclBinding.entry().operation().toString().equals("WRITE")
+            || aclBinding.entry().operation().toString().equals("READ")) {
+          acls.add(aclbindingMap);
+        }
```

Comment:
I guess `aclBinding.entry().operation().toString()` and `aclBinding.pattern().resourceType().toString()` could be extracted into vars since they are used more than once here